File: phone/src/protocol/models/media.dart
class MediaModel {
  const MediaModel({
    required this.mediaId,
    required this.mediaType,
    this.codec,
    this.format,
    this.sampleRate,
    this.channels,
    this.width,
    this.height,
    this.durationMs,
    this.frameIndex,
    this.chunkIndex,
    this.isFinal,
    this.capturedAt,
    this.payloadRef,
    this.metadata = const {},
  });

  final String mediaId;
  final String mediaType;
  final String? codec;
  final String? format;
  final int? sampleRate;
  final int? channels;
  final int? width;
  final int? height;
  final int? durationMs;
  final int? frameIndex;
  final int? chunkIndex;
  final bool? isFinal;
  final String? capturedAt;
  final String? payloadRef;
  final Map<String, dynamic> metadata;

  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'media_id': mediaId,
      'media_type': mediaType,
      'codec': codec,
      'format': format,
      'sample_rate': sampleRate,
      'channels': channels,
      'width': width,
      'height': height,
      'duration_ms': durationMs,
      'frame_index': frameIndex,
      'chunk_index': chunkIndex,
      'is_final': isFinal,
      'captured_at': capturedAt,
      'payload_ref': payloadRef,
      'metadata': metadata,
    };
  }
}

File: server/src/protocol/models/media.py
from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any

from protocol.enums import MediaType
from protocol.models.base import Serializable


@dataclass(slots=True)
class MediaModel(Serializable):
    media_id: str
    media_type: MediaType
    codec: str | None = None
    format: str | None = None
    sample_rate: int | None = None
    channels: int | None = None
    width: int | None = None
    height: int | None = None
    duration_ms: int | None = None
    frame_index: int | None = None
    chunk_index: int | None = None
    is_final: bool | None = None
    captured_at: str | None = None
    payload_ref: str | None = None
    metadata: dict[str, Any] = field(default_factory=dict)

    @classmethod
    def from_dict(cls, raw: dict[str, Any]) -> "MediaModel":
        return cls(
            media_id=raw["media_id"],
            media_type=MediaType(raw["media_type"]),
            codec=raw.get("codec"),
            format=raw.get("format"),
            sample_rate=raw.get("sample_rate"),
            channels=raw.get("channels"),
            width=raw.get("width"),
            height=raw.get("height"),
            duration_ms=raw.get("duration_ms"),
            frame_index=raw.get("frame_index"),
            chunk_index=raw.get("chunk_index"),
            is_final=raw.get("is_final"),
            captured_at=raw.get("captured_at"),
            payload_ref=raw.get("payload_ref"),
            metadata=dict(raw.get("metadata") or {}),
        )
